vlog4j-parser/src/main/java/org/semanticweb/vlog4j/parser/RuleParser.java

```diff
@@ -70,6 +71,16 @@ public static KnowledgeBase parse(String input) throws ParsingException {
 		return parse(inputStream, "UTF-8");
 	}
 
+	public static Rule parseRule(String input) throws ParsingException {
+		InputStream inputStream = new ByteArrayInputStream(input.getBytes());
+		JavaCCParser localParser = new JavaCCParser(inputStream, "UTF-8");
+		try {
+			return localParser.rule();
+		} catch (ParseException | PrefixDeclarationException | TokenMgrError e) {
+			throw new ParsingException(e.getMessage(), e);
```

Comment:
An error should be logged before throwing the exception. Also, the exception message should contain more info.
If you pass the exception e as a parameter to the ParsingException, you don't need to propagate e.getMessage(): it will be automatically propagated (see below). Instead, you should write an explicit message like "Error during rule parsing".


The following program outputs the lines below.

`public static void main(final String[] args) throws Exception {
		try {
			throw new Exception("a");
		} catch (final Exception e) {
			throw new Exception("b", e);
		}
	}`

Exception in thread "main" java.lang.Exception: b
	at playground.Main.main(Main.java:9)
Caused by: java.lang.Exception: a
	at playground.Main.main(Main.java:7)

Reply:
please implement these changes in all relevant methods in this class.